pkg/odo/cli/url/delete.go

```diff
@@ -41,7 +40,7 @@ func NewURLDeleteOptions() *URLDeleteOptions {
 // Complete completes URLDeleteOptions after they've been Deleted
 func (o *URLDeleteOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
 
-	if experimental.IsExperimentalModeEnabled() && util.CheckPathExists(o.DevfilePath) {
```

Comment:
removing this check will make `odo url delete` not work for s2i components
```
[mrinaldas@localhost project]$ odo create nodejs --s2i
Experimental mode is enabled, use at your own risk

Validation
 ✓  Validating component [1s]

Please use `odo push` command to create the component with source deployed
[mrinaldas@localhost project]$ odo url create example
 ✓  URL example created for component: nodejs-project-vwui

To apply the URL configuration changes, please use `odo push`
[mrinaldas@localhost project]$ odo url delete example
 ✗  The current directory does not represent an odo component. Use 'odo create' to create component here or switch to directory with a component
```

Reply:
updated